rosbag2_transport/CMakeLists.txt

```diff
@@ -149,45 +159,68 @@ if(BUILD_TESTING)
       yaml_cpp_vendor)
   endif()
 
-  ament_add_gmock(test_formatter
-    test/rosbag2_transport/test_formatter.cpp
-    src/rosbag2_transport/formatter.cpp
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
-  if(TARGET test_formatter)
-    target_link_libraries(test_formatter rosbag2_transport)
-  endif()
-
   # disable the following tests for connext
   # due to slower discovery of nodes
-  get_default_rmw_implementation(rmw_default)
   set(SKIP_TEST "")
-  if(${rmw_default} MATCHES "(.*)connext(.*)")
+  if(${rmw_implementation} MATCHES "(.*)connext(.*)")
     set(SKIP_TEST "SKIP_TEST")
   endif()
 
-  ament_add_gmock(test_record
+  set(test_record test_record${target_suffix})
+  ament_add_gmock(${test_record}
     test/rosbag2_transport/test_record.cpp
+    ENV ${rmw_implementation_env_var}
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
     ${SKIP_TEST})
-  if(TARGET test_record)
-    target_link_libraries(test_record rosbag2_transport)
-    ament_target_dependencies(test_record test_msgs rosbag2_test_common)
+  if(TARGET ${test_record})
+    target_link_libraries(${test_record} rosbag2_transport)
+    ament_target_dependencies(${test_record}
+      ${rmw_implementation} test_msgs rosbag2_test_common)
   endif()
 
-  ament_add_gmock(test_play
+  set(test_play test_play${target_suffix})
+  ament_add_gmock(${test_play}
     test/rosbag2_transport/test_play.cpp
+    ENV ${rmw_implementation_env_var}
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
     ${SKIP_TEST})
-  if(TARGET test_play)
-    target_link_libraries(test_play rosbag2_transport)
-    ament_target_dependencies(test_play test_msgs rosbag2_test_common)
+  if(TARGET ${test_play})
+    target_link_libraries(${test_play} rosbag2_transport)
+    ament_target_dependencies(${test_play}
+      ${rmw_implementation} test_msgs rosbag2_test_common)
+  endif()
+endfunction()
+
+macro(test_rmw_implementation_wrapper)
```

Comment:
macro -> function? Do you need to export things out of the scope of the macro? If not, a function is preferable as it defines a scope and avoids naming clash.

Reply:
was able to  skip this entirely. i was copying `rcl`'s tests but they actually import the rmw implementation, we don't have to